cmd/restic/cmd_backup.go

```diff
@@ -522,3 +544,87 @@ func readExcludePatternsFromFiles(excludeFiles []string) []string {
 	}
 	return excludes
 }
+
+// FilenameCheck is a function that takes a filename and returns a boolean depending on arbitraty check
+type FilenameCheck func(filename string) bool
+
+// genExcludeByFile returns a FilenameCheck which itself returns whether a path
+// should be excluded. The FilenameCheck considers a file to be excluded when
+// it resides in a directory with an exclusion file, that is either specified
+// by excludeIfPresent in the form "filename[:content]".  The returned error is
+// non-nil if the filename component of excludeIfPresent is empty.
+func genExcludeByFile(excludeIfPresent string) (FilenameCheck, error) {
+	fn := func(string) bool { return false }
+	tf, tc := "", ""
+	if excludeIfPresent != "" {
+		colon := strings.Index(excludeIfPresent, ":")
+		switch {
+		default:
+			tf = excludeIfPresent[:colon]
+			tc = excludeIfPresent[colon+1:]
+		case colon < 0:
+			tf = excludeIfPresent
+		case colon == 0:
+			return nil, fmt.Errorf("no name for exclusion tagfile provided")
+		}
+	}
+	if tf != "" {
```

Comment:
This case cannot happen (you've handled `colon == 0` earlier already), drop the `if`.

Reply:
Done.